### PR TITLE
Support start time

### PR DESCRIPTION
**Pull request by JuanitoFatas**
Hello 👋 

I think it will be useful to embed a video that starts from a specific time to save users‘s time.

### CODE REVIEW

File: src/lite-yt-embed.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ class LiteYTEmbed extends HTMLElement {
         // Gotta encode the untrusted value
         // https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#rule-2---attribute-escape-before-inserting-untrusted-data-into-html-common-attributes
         this.videoId = encodeURIComponent(this.getAttribute('videoid'));
-
+        this.startTime = encodeURIComponent(this.getAttribute('startTime'));
```
Comment (Garbee) [inline src/lite-yt-embed.js]: Wouldn't we want to `parseInt` the value? That would still give a clear direct value being filtered for security while also validating what we give to the embed URL is a number.

File: src/lite-yt-embed.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ class LiteYTEmbed extends HTMLElement {
         const iframeHTML = `
 <iframe width="560" height="315" frameborder="0"
   allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen
-  src="https://www.youtube.com/embed/${this.videoId}?autoplay=1"
+  src="https://www.youtube.com/embed/${this.videoId}?autoplay=1&start=${this.startTime}"
```
Comment (Garbee) [inline src/lite-yt-embed.js]: I kinda feel icky about adding parameters that may not even have a value to the URL. I think I'd wait a bit on pulling this in until my work on actively listening for attribute changes is done this weekend. Once that lands we can do a small refactor of the URL where it is created using the URL object allowing for trivial dynamic creation of the string based on the present attributes. Then supporting any given attributes YouTube allows should be a very thin layer to add on top in a repeatable pattern.

**Comment by Garbee** [comment]
Another note on this as @laukstein points out in #30, `startTime` is not a valid attribute since it contains an uppercase character. 

`start-time` would perhaps be a better attribute name or simply `start`? I think `start` and `end` for those times would be better attribute names, but I'll leave picking whether to have `-time` appended on up to Paul or anyone else with a strong opinion on that matter.

File: src/lite-yt-embed.js [hunk 3]
```diff
@@ -102,7 +102,7 @@ class LiteYTEmbed extends HTMLElement {
         const iframeHTML = `
 <iframe width="560" height="315" frameborder="0"
   allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen
-  src="https://www.youtube.com/embed/${this.videoId}?autoplay=1"
+  src="https://www.youtube.com/embed/${this.videoId}?autoplay=1&start=${this.start}"
```
Comment (Garbee) [inline src/lite-yt-embed.js]: So... thought about this fresh this morning... We do need to wait on getting this support until we have better attribute observation support and dynamic query param generation. With the current flow and this being "always added" it is easy to get `"null", "undefined", or "NaN"` added into this parameter. While I *believe* YouTube would be guarded enough to just play from the beginning in that case, I still would err on the side of not sending known broken data around.

**Comment by paulirish** [comment]
i'd like to be conservative on adding features. there's ~20 parameters like this that could be added. 

what is the usecase for an embed using starttime? 

if you told me that discord was going to use this embed and wanted to translate a YT link (with `ts` param`) into this embed, then sure we could add this feature. but i don't think it's a clear win for everyone.

**Comment by Garbee** [comment]
Any given blog post author could want to use the element to speed up their pages. Blog posts which could be linked to specific sections of content within videos for context on what the author is discussing.

That's just one (albeit no super major) use case. Once the attribute observation stuff is finished it would be relatively easy to support as many parameters as we'd want. The main concern is going to be the total file size/weight we'd want the resulting JS to be.

**Comment by JuanitoFatas** [comment]
My use case is to use [on blog post](https://juanitofatas.com/fragments/lite-youtube-embed) to save readers time as @Garbee mentioned. 

My point has been made and I’m closing this PR.

Thanks everyone!

**Comment by paulirish** [comment]
lol that is a compelling usecase. 😳 linking into my presentation where i debut this exact project. 

damn it.

**Pull request closed without merging**